### DISCUSSION TITLE
DAG expressiveness

### DISCUSSION THREAD

**Discussion by domhues** [Netflix/conductor#3486] [Q&A]
Hi everybody, I am trying to get myself familiar with the expressiveness of the workflow language in conductor compared to a generic DAG and I am struggling with defining following type of DAG with conductor's fork join logic:
![Unbenanntes Diagramm drawio (80)](https://user-images.githubusercontent.com/70632459/218439501-58315e2c-4836-466e-90d1-526c919bb58a.png)
I.e. Task 3 should only start when Task 1 and 2 finished, but Task 4 should already be able to start if task 2 finishes while task 4 potentially still running.
I stumbled across partial join operators as explained in [here](https://conductor.netflix.com/documentation/configuration/workflowdef/operators/fork-task.html) but they did not fit the use-case either. Could you point me to a documentation about the expressiveness or hint into the direction which enables these kind of DAGs?

Thanks a lot!

**Comment by manan164** [comment]
Hi @domhues , I think the below workflow should suffice the use case. Let me know if this works, or we can chat [here](https://join.slack.com/t/orkes-conductor/shared_invite/zt-xyxqyseb-YZ3hwwAgHJH97bsrYRnSZg) for more realtime collaboration.
<img width="1513" alt="Screenshot 2023-02-14 at 7 50 48 PM" src="https://user-images.githubusercontent.com/1897158/218765792-f13f7f04-f075-4fdb-b69a-5234f9880740.png">

**Reply by domhues** [reply]
@manan164 Thanks a lot for your answer! 
I think your solution does indeed allows Task 4 to execute when Task 2 is done, independent of Task 1 still running.
However, it also adds a new dependency between Task 4 and Task 3, correct? Task 3 is unnecessarily delayed by task 4 while in the DAG [above ](https://github.com/Netflix/conductor/discussions/3486#discussion-4852755) task 3 and 4 can run in parallel.
